plugins/download.py

```diff
@@ -128,9 +146,34 @@ def _download_debuginfo(self, pkg_specs):
                 if dbg_found:
                     break
 
-        self.base.download_packages(dbg_pkgs, self.base.output.progress)
-        locations = sorted([pkg.localPkg() for pkg in dbg_pkgs])
-        return locations
+        return dbg_pkgs
+
+    @staticmethod
+    def _print_download_urls_for_pkgs(pkgs):
+        """Print out a url for each dnf.Package object in the pkgs list."""
+
+        # Create a tmpdir so that librepo tmp files don't get left behind
+        with dnf.util.tmpdir() as tmpdir:
+            # For each dnf.Package object get the librepo.Handle (which
+            # represents a download configuration) and set fetchmirrors
+            # so that it will only get a mirrorlist when calling perform()
+            # and won't actually try to download anything else. Once we
+            # have a mirrorlist we can then just choose one of the mirrors
+            # and merge the mirror url with the relative location of the
+            # package on the mirror.
+            for pkg in pkgs:
+                h = pkg.repo._get_handle()
+                h.fetchmirrors = True  # sets LRO_FETCHMIRRORS
+                h.destdir = tmpdir     # store tmp files in tmpdir
+                h.perform()
+                for mirror in h.mirrors:
+                    if mirror.startswith('https'):
```

Comment:
oops, I originally had this as http, not https. but I decided to try out https and forgot to change it back. Will fix it to http.

My thoughts about ftp and rsync is that they are generally less useful than the http(s) ones. almost everything can use http(s). The alternative to this is that we give some option for the user to specify what protocol they want. I think that is overcomplicating it though. What do you think?